dashboard/test/models/user_test.rb

```diff
@@ -541,6 +541,45 @@ class UserTest < ActiveSupport::TestCase
     assert_equal(1, user.next_unpassed_progression_level(script).chapter)
   end
 
+  test 'can get next_unpassed_progression_level when last updated user_level is inside a level group' do
+    user = create :user
+    script = create :script
+
+    sub_level1 = create :text_match, name: 'sublevel1'
+    create :text_match, name: 'sublevel2'
+
+    level_group = create :level_group, name: 'LevelGroupLevel1', type: 'LevelGroup'
+    level_group.properties['pages'] = [{levels: ['level_multi1', 'level_multi2']}]
+
+    create(:script_level, script: script, levels: [level_group])
+    create :user_script, user: user, script: script
+
+    # Create a UserLevel for our level_group and sublevel, the sublevel is more recent
+    user_level1 = UserLevel.create(
+      user: user,
+      level: level_group,
+      script: script,
+      attempts: 1,
+      best_result: Activity::MINIMUM_PASS_RESULT,
+      updated_at: Time.now - 1
+    )
+
+    user_level2 = UserLevel.create(
+      user: user,
+      level: sub_level1,
+      script: script,
+      attempts: 1,
+      best_result: Activity::MINIMUM_PASS_RESULT,
+      updated_at: Time.now
+    )
+
+    assert(user_level1.updated_at < user_level2.updated_at)
```

Comment:
Sometimes I see value in asserting what's important about the test data I've constructed. Yes, it should never fail, but it also makes it so that if I'm debugging why a later assertion did fail, I can be certain about this particular assumption.